scripts/actions/spells/white/sacrifice.lua

```diff
@@ -10,25 +11,48 @@ end
 spellObject.onSpellCast = function(caster, target, spell)
     local removables = { xi.effect.FLASH, xi.effect.BLINDNESS, xi.effect.PARALYSIS, xi.effect.POISON, xi.effect.CURSE_I, xi.effect.CURSE_II, xi.effect.DISEASE, xi.effect.PLAGUE }
 
-    -- remove one effect and add it to me
-    for i, effect in ipairs(removables) do
-        if target:hasStatusEffect(effect) then
-            spell:setMsg(xi.msg.basic.MAGIC_ABSORB_AILMENT)
+    -- Maximum number of effects to transfer
+    local maxTransferCount = caster:hasStatusEffect(xi.effect.AFFLATUS_SOLACE) and 7 or 1
+
+    -- Use the helper function to transfer status effects
+    local transferCount = 0  -- Initialize transferCount
 
-            local statusEffect = target:getStatusEffect(effect)
+    -- Loop over the original effects
+    for _, effect in ipairs(removables) do
+        if transferCount < maxTransferCount then
+            if target:hasStatusEffect(effect) then
```

Comment:
Wouldn't it make more sense here to iterate over the target's effects as opposed to a list of possible effects for matching purposes?

Reply:
I can try to do that!

Reply:
It looks like I can use target:getStatusEffects() to iterate over the target's effects. I can try my best to implement this change if we want the solution to be best practice. I was proud to get this working in the first place!

Reply:
I'm having a little bit of trouble getting this to work, I'm gonna keep trying